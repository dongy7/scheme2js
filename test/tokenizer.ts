import Compiler from '../src/compiler'

it('tokenizes number', () => {
  expect(Compiler.tokenizer('123')).toMatchSnapshot()
})

it('tokenizes negative number', () => {
  expect(Compiler.tokenizer('-123')).toMatchSnapshot()
})

it('tokenizes parens', () => {
  expect(Compiler.tokenizer('()')).toMatchSnapshot()
})

it('tokenizes names', () => {
  expect(Compiler.tokenizer('foo')).toMatchSnapshot()
})

it('tokenizes a list', () => {
  expect(Compiler.tokenizer('(123 foo (bar baz))')).toMatchSnapshot()
})
