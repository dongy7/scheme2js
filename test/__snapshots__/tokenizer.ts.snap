// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenizes a list 1`] = `
Array [
  Object {
    "type": "paren",
    "value": "(",
  },
  Object {
    "type": "number",
    "value": "123",
  },
  Object {
    "type": "name",
    "value": "foo",
  },
  Object {
    "type": "paren",
    "value": "(",
  },
  Object {
    "type": "name",
    "value": "bar",
  },
  Object {
    "type": "name",
    "value": "baz",
  },
  Object {
    "type": "paren",
    "value": ")",
  },
  Object {
    "type": "paren",
    "value": ")",
  },
]
`;

exports[`tokenizes names 1`] = `
Array [
  Object {
    "type": "name",
    "value": "foo",
  },
]
`;

exports[`tokenizes number 1`] = `
Array [
  Object {
    "type": "number",
    "value": "123",
  },
]
`;

exports[`tokenizes parens 1`] = `
Array [
  Object {
    "type": "paren",
    "value": "(",
  },
  Object {
    "type": "paren",
    "value": ")",
  },
]
`;
